src/btree/row_modify.c

```diff
@@ -113,6 +113,18 @@ __wt_row_modify(WT_CURSOR_BTREE *cbt, const WT_ITEM *key, const WT_ITEM *value,
             /* Avoid WT_CURSOR.update data copy. */
             __wt_upd_value_assign(cbt->modify_update, upd);
         } else {
+            /*
+             * We only update history store records in two cases:
+             *  1) Delete the record with a tombstone with WT_TS_NONE.
+             *  2) Update the record's stop time point if the prepared update written to the data
+             * store is committed.
+             */
+            WT_ASSERT(session,
+              !WT_IS_HS(S2BT(session)) ||
+                (upd_arg->type == WT_UPDATE_TOMBSTONE && upd_arg->start_ts == WT_TS_NONE &&
+                  upd_arg->next == NULL) ||
+                (upd_arg->type == WT_UPDATE_TOMBSTONE && upd_arg->next != NULL &&
+                  upd_arg->next->type == WT_UPDATE_STANDARD && upd_arg->next->next == NULL));
```

Comment:
To verify we only do this if previously we don't have a stop time point requires cracking the cell. I don't think we need to go that far.